DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeIDGenerator.swift

```diff
@@ -32,28 +33,32 @@ internal final class NodeIDGenerator {
 
     /// Returns single `NodeID` for given instance of `UIView`.
     /// - Parameter view: the `UIView` object
+    /// - Parameter nodeRecorder: the `NodeRecorder` object
     /// - Returns: the `NodeID` of queried instance
-    func nodeID(for view: UIView) -> NodeID {
-        if let currentID = view.nodeID {
-            return currentID
+    func nodeID(view: UIView, nodeRecorder: NodeRecorder) -> NodeID {
+        let key = nodeRecorder.identifier.uuidString + "-\(view.hash)"
+        if let nodeID = ids[key]?.first {
+            return nodeID
         } else {
-            let id = getNextID()
-            view.nodeID = id
-            return id
+            let nodeID = getNextID()
+            ids[key] = [nodeID]
+            return nodeID
         }
     }
 
     /// Returns multiple `NodeIDs` for given instance of `UIView`.
     /// - Parameter size: the number of IDs
     /// - Parameter view: the `UIView` object
+    /// - Parameter nodeRecorder: the `NodeRecorder` object
     /// - Returns: an array with given number of `NodeID` values
-    func nodeIDs(_ size: Int, for view: UIView) -> [NodeID] {
-        if let currentIDs = view.nodeIDs, currentIDs.count == size {
-            return currentIDs
+    func nodeIDs(_ size: Int, view: UIView, nodeRecorder: NodeRecorder) -> [NodeID] {
+        let key = nodeRecorder.identifier.uuidString + "-\(view.hash)-\(size)"
```

Comment:
String concatenation and interpolation are known to be extremely slow. What is the overhead we're adding with this change? Keep in mind this will be called for each view in view-tree, every time we capture a `record()`.